third_party/terraform/resources/resource_storage_bucket.go

```diff
@@ -349,15 +349,7 @@ func resourceStorageBucketCreate(d *schema.ResourceData, meta interface{}) error
 	}
 
 	if v, ok := d.GetOk("retention_policy"); ok {
-		retention_policies := v.([]interface{})
-
-		sb.RetentionPolicy = &storage.BucketRetentionPolicy{}
-
-		retentionPolicy := retention_policies[0].(map[string]interface{})
-
-		if v, ok := retentionPolicy["retention_period"]; ok {
-			sb.RetentionPolicy.RetentionPeriod = int64(v.(int))
-		}
+		sb.RetentionPolicy = expandBucketRetentionPolicy(v.([]interface{}))
```

Comment:
This won't actually work because you can't set the value of is_locked on create (it's probably worth adding a comment that says so, so we don't wonder about it in the future).

test output:
```
------- Stdout: -------
=== RUN   TestAccStorageBucket_retentionPolicyLocked
=== PAUSE TestAccStorageBucket_retentionPolicyLocked
=== CONT  TestAccStorageBucket_retentionPolicyLocked
Error creating bucket tf-test-acc-bucket-250558721639430483: googleapi: Error 400: Retention Policy should not set/unset lock at insertion time for bucket 'tf-test-acc-bucket-250558721639430483'. Use lockRetentionPolicy rpc endpoint to lock retention policy explicitly., invalid--- FAIL: TestAccStorageBucket_retentionPolicyLocked (1.62s)
    testing.go:635: Step 0 error: errors during apply:
        
        Error: googleapi: Error 400: Retention Policy should not set/unset lock at insertion time for bucket 'tf-test-acc-bucket-250558721639430483'. Use lockRetentionPolicy rpc endpoint to lock retention policy explicitly., invalid
        
          on /opt/teamcity-agent/temp/buildTmp/tf-test637902388/main.tf line 2:
          (source code not available)
        
        
FAIL
```

Reply:
Done.